src/lib.rs

```diff
@@ -544,6 +705,135 @@ mod test {
+                // Chekcing if knapsack selects exactly 2 inputs
+                assert_eq!(result.selected_inputs.len(), 2);
+                // Checking if the selected inputs are 5, 2 CENTS
+                inputs_verify = vec![0,2];
+                assert!(inputs_verify.iter().all(|&item| result.selected_inputs.contains(&item)));
+            }
+            // Testing if knapsack can select 3 inputs (5,2,1) CENTS to make 8 CENTS 
+            inputs_verify.clear();
+            options = setup_options((8.0 * CENT).round() as u64);
+            if let Ok(result) = select_coin_knapsack(&inputs, options) {
+                // Chekcing if knapsack selects exactly 3 inputs
+                assert_eq!(result.selected_inputs.len(), 3);
+                // Checking if the selected inputs are 5,2,1 CENTS
+                inputs_verify = vec![0,2,1];
+                assert!(inputs_verify.iter().all(|&item| result.selected_inputs.contains(&item)));
+            }
+            // Testing if knapsack can select 1 input (10) CENTS to make 9 CENTS 
+            inputs_verify.clear();
+            options = setup_options((10.0 * CENT).round() as u64);
+            if let Ok(result) = select_coin_knapsack(&inputs, options) {
+                // Chekcing if knapsack selects exactly 1 inputs
+                assert_eq!(result.selected_inputs.len(), 1);
+                // Checking if the selected inputs are 10 CENTS
+                inputs_verify = vec![10];
+                assert!(inputs_verify.iter().all(|&item| result.selected_inputs.contains(&item)));
+            }
+            // Clearing the input vector
+            inputs.clear();
+            inputs_verify.clear();
+            // Adding 30, 20, 8, 7,6 CENT to the wallet, totalling 71 CENTS
+            // Adding 0.001 CENT to the two inputs to account for fees
+            inputs = setup_output_groups(
+                vec![
+                    (6.001 * CENT).round() as u64,
+                    (7.001 * CENT).round() as u64,
+                    (8.001 * CENT).round() as u64,
+                    (20.001 * CENT).round() as u64,
+                    (30.001 * CENT).round() as u64,
+                ],
+                vec![100, 200, 100, 10, 5],
+            );
+            // Test if Knapsack retruns an Error while trying to select inputs totalling 72 CENTS
+            options = setup_options((72.0 * CENT).round() as u64);
+            result = select_coin_knapsack(&inputs, options);
+            assert!(matches!(result, Err(SelectionError::NoSolutionFound)));
+            // Testing if knapsack can select 1 input (20) CENTS to make 16 CENTS 
+            inputs_verify.clear();
+            options = setup_options((16.0 * CENT).round() as u64);
+            if let Ok(result) = select_coin_knapsack(&inputs, options) {
+                // Chekcing if knapsack selects exactly 1 inputs
+                assert_eq!(result.selected_inputs.len(), 1);
```

Comment:
This assertion is wrong because knapsack tries to select inputs which are lower than the target.
So you should assert for length to be 3